### ISSUE TITLE
Some key combinations cause panic

### ISSUE THREAD

**Issue by gyscos** [ihalila/pancurses#4]
Simple program to try input support:
```rust
extern crate pancurses;

fn main() {
    let pc = pancurses::initscr();
    pc.keypad(true);
    loop {
        if let Some(ev) = pc.getch() {
            println!("{:?}", ev);
        }
    }
}
```

Pressing `Ctrl-Left` (and some other combinations) results in a panic:

```
thread 'main' panicked at 'index out of bounds: the len is 108 but the index is 243', /home/gyscos/.cargo/registry/src/github.com-1ecc6299db9ec823/pancurses-0.4.0/src/unix/mod.rs:66
stack backtrace:
   1:     0x555e3f9df4ff - std::sys::backtrace::tracing::imp::write::h22f199c1dbb72ba2
   2:     0x555e3f9e1fad - std::panicking::default_hook::{{closure}}::h9a389c462b6a22dd
   3:     0x555e3f9e150a - std::panicking::default_hook::h852b4223c1c00c59
   4:     0x555e3f9e1a58 - std::panicking::rust_panic_with_hook::hcd9d05f53fa0dafc
   5:     0x555e3f9e18f2 - std::panicking::begin_panic::hf6c488cee66e7f17
   6:     0x555e3f9e1830 - std::panicking::begin_panic_fmt::hb0a7126ee57cdd27
   7:     0x555e3f9e17b1 - rust_begin_unwind
   8:     0x555e3fa16d9f - core::panicking::panic_fmt::h9af671b78898cdba
   9:     0x555e3fa16d43 - core::panicking::panic_bounds_check::h56f656aa4e352200
  10:     0x555e3f9d7840 - pancurses::unix::to_special_keycode::hb00b80a8963e12bc
                        at /home/gyscos/.cargo/registry/src/github.com-1ecc6299db9ec823/pancurses-0.4.0/src/unix/mod.rs:66
  11:     0x555e3f9d78e8 - pancurses::Window::getch::h179489f0c955b58f
                        at /home/gyscos/.cargo/registry/src/github.com-1ecc6299db9ec823/pancurses-0.4.0/src/lib.rs:178
  12:     0x555e3f9d6c2e - pan::main::hecb42b9052c3a41e
                        at /home/gyscos/test/pan/src/main.rs:7
  13:     0x555e3f9e9a76 - __rust_maybe_catch_panic
  14:     0x555e3f9e0d81 - std::rt::lang_start::h14cbded5fe3cd915
  15:     0x555e3f9d6d43 - main
  16:     0x7fcca047d290 - __libc_start_main
  17:     0x555e3f9d6a09 - _start
  18:                0x0 - <unknown>
```

Tested using pancurses 0.4.0 on linux.

**Comment by ihalila** [comment]
Thanks for the bug report, I'm fairly sure I know what's causing that too.
I'll get on it.

**Comment by ihalila** [comment]
This ought to be fixed now, I released [0.4.1](https://github.com/ihalila/pancurses/releases/tag/0.4.1) with a fix.
